packages/gguf/src/types.ts

```diff
@@ -0,0 +1,141 @@
+export type MetadataBaseValue = string | number | bigint | boolean;
+export type MetadataValue = MetadataBaseValue | MetadataBaseValue[] | MetadataValue[]; /// recursive as arrays can be nested.
+
+export type Version = 1 | 2 | 3;
+
+export enum GGMLQuantizationType {
+	F32 = 0,
+	F16 = 1,
+	Q4_0 = 2,
+	Q4_1 = 3,
+	Q5_0 = 6,
+	Q5_1 = 7,
+	Q8_0 = 8,
+	Q8_1 = 9,
+	Q2_K = 10,
+	Q3_K = 11,
+	Q4_K = 12,
+	Q5_K = 13,
+	Q6_K = 14,
+	Q8_K = 15,
+	IQ2_XXS = 16,
+	IQ2_XS = 17,
+	IQ3_XXS = 18,
+	IQ1_S = 19,
+	IQ4_NL = 20,
+	IQ3_S = 21,
+	IQ2_S = 22,
+	IQ4_XS = 23,
+}
+
+export enum GGUFValueType {
+	UINT8 = 0,
+	INT8 = 1,
+	UINT16 = 2,
+	INT16 = 3,
+	UINT32 = 4,
+	INT32 = 5,
+	FLOAT32 = 6,
+	BOOL = 7,
+	STRING = 8,
+	ARRAY = 9,
+	UINT64 = 10,
+	INT64 = 11,
+	FLOAT64 = 12,
+}
+
+export const ARCHITECTURES = [
+	"llama",
+	"mpt",
+	"gptneox",
+	"gptj",
+	"gpt2",
+	"bloom",
+	"falcon",
+	"gemma",
+	"rwkv",
+	"whisper",
```

Comment:
```suggestion
    "llama",
    "falcon",
    "baichuan",
    "gpt2",
    "gptj",
    "gptneox",
    "mpt",
    "starcoder",
    "persimmon",
    "refact",
    "bert",
    "nomic-bert",
    "bloom",
    "stablelm",
    "qwen",
    "qwen2",
    "phi2",
    "plamo",
    "codeshell",
    "orion",
    "internlm2",
    "minicpm",
    "gemma",
    "starcoder2",
    "mamba",
```

Reply:
(optional, but it's the current list from the llama.cpp source of truth IIUC)

Reply:
great ! Will update them and their specific types from (ex: https://github.com/ggerganov/ggml/blob/master/docs/gguf.md#mamba)